afl/build.rs

```diff
@@ -44,12 +58,16 @@ fn main() {
         PathBuf::from(AFL_SRC_PATH)
     };
 
-    let base = if env::var("DOCS_RS").is_ok() {
+    let base = if building_on_docs_rs {
         Some(PathBuf::from(out_dir))
     } else {
         None
     };
 
+    // smoelius: Lock `work_dir` until the build script exits.
+    #[cfg(unix)]
+    let _file = sys::lock_path(&work_dir).unwrap();
```

Comment:
Just making sure I understand: this doesn't require an associated `flock(..., LOCK_UN)` because you're expecting the advisory lock to be dropped when the underlying fd is closed at process exit, right?

I believe that should be fine, but it's not super well defined by POSIX AFAICT. Just something to be aware of should it cause unusual behavior on non-Linux builds in the future 🙂

Reply:
> Just making sure I understand: this doesn't require an associated `flock(..., LOCK_UN)` because you're expecting the advisory lock to be dropped when the underlying fd is closed at process exit, right?

That's correct.

> I believe that should be fine, but it's not super well defined by POSIX AFAICT. Just something to be aware of should it cause unusual behavior on non-Linux builds in the future 🙂

I'm curious as to what makes you say that (not saying you're wrong).

I'm reading this: https://man7.org/linux/man-pages/man2/flock.2.html

> Furthermore, the lock is released
       either by an explicit LOCK_UN operation on any of these duplicate
       file descriptors, or when all such file descriptors have been
       closed.

Is file descriptor closure not well defined?

Reply:
> Is file descriptor closure not well defined?

I thought POSIX was ambiguous on whether process termination is actually required to close all FDs, but it looks like it's unambiguous:

> All of the file descriptors, directory streams, conversion descriptors, and message catalog descriptors open in the calling process shall be closed.

(From <https://pubs.opengroup.org/onlinepubs/007904875/functions/exit.html>)

There's some subtlety around child processes that inherit the open FD, but I don't think that case matters here. So this is safe to ignore 🙂